lib/block_view_validator.go

```diff
@@ -1635,6 +1802,16 @@ func (bav *UtxoView) IsValidRegisterAsValidatorMetadata(
 		}
 	}
 
+	// Error if VotingPublicKey is already taken.
+	validatorBLSPublicKeyPKIDPairEntry, err := bav.GetBLSPublicKeyPKIDPairEntry(metadata.VotingPublicKey)
```

Comment:
I'm not going throught his BLS public key manipulation too closely. Just test it thoroughly to make sure it's all updating correctly across epoch boundaries especially.